Integrate Simulator v0.1-Save Simulation Parameters

stdevi opened the pull request:
Closes AP-1874

Epic link: https://apromore.atlassian.net/l/c/B5Q8EQct

Commit:
feat(QBP Simulator): Integrate Simulator v0.1-Save Simulation Parameters

Closes AP-1874

Commit:
feat(QBP Simulator): Integrate Simulator v0.1-Save Simulation Parameters

Closes AP-1874

Commit:
feat(QBP Simulator): Integrate Simulator v0.1-Save Simulation Parameters

Closes AP-1874

Commit:
feat(QBP Simulator): Integrate Simulator v0.1-Save Simulation Parameters

Closes AP-1874

Commit:
feat(QBP Simulator): Integrate Simulator v0.1-Save Simulation Parameters

Closes AP-1874

Commit:
feat(QBP Simulator): Integrate Simulator v0.1-Save Simulation Parameters

Closes AP-1874

Commit:
Squashed commit of the following:

commit 92c91f8cf8b387c33778439f969671325dbd5701
Merge: db95b0056 a33eebbe3
Author: dreamind <dreamind@gmail.com>
Date:   Sat Oct 24 20:57:01 2020 +1100

    Merge pull request #529 from apromore/feature/AP-2120-2432

    AP-2120-2432 Merge and PD shortcuts fixes

commit a33eebbe39fe639db160d450d9e4fa68b06c8d06
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Sat Oct 24 20:55:30 2020 +1100

    AP-2120-2432 Merge and PD shortcuts fixes

commit db95b00566b9675c2c89c82bc0b1cc9b9506fcc8
Merge: f39f5ebea 6812f38a4
Author: frankma <49012776+frank-ma-leonardo@users.noreply.github.com>
Date:   Thu Oct 22 12:29:04 2020 +1100

    Merge pull request #525 from apromore/AP-2431

    APIs to get dependent artifact of user metadata

commit 6812f38a48031ccc5b839dc43cd026cf04dc85d2
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Thu Oct 22 09:49:16 2020 +1100

    Add method to get the usermetadata AccessType

commit 1e4e08b88ffe540b4d2c98c4fd2b2dab130ec16d
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 21 17:41:39 2020 +1100

    Remove unnecessary assertions

commit f39f5ebea69de65982ee0a06274c40ffe73cd06e
Merge: 41e0d2198 e02129fc3
Author: raboczi <raboczi@icloud.com>
Date:   Wed Oct 21 15:12:05 2020 +1000

    Merge branch 'feature/AP-2416-hadoop-on-windows' into development

commit 41e0d2198619db71302aa74d7a5fc17c3d1e4cf6
Merge: 915b7bf09 50f7b6c95
Author: dreamind <dreamind@gmail.com>
Date:   Wed Oct 21 11:22:48 2020 +1100

    Merge pull request #526 from apromore/feature/AP-2236

    AP-2236 Fix long graph search result in PD

commit 50f7b6c957df9a4bdf55620540ad35009ec9dbec
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Wed Oct 21 11:21:22 2020 +1100

    AP-2236 Fix long graph search result in PD

commit a1ecf9ff415bf86a065214c145ae089f7d728afc
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 21 09:06:13 2020 +1100

    APIs to get dependent artifact of user metadata

commit 915b7bf095e6c9f547d162ff5d077bc4af6e2959
Merge: ba4a3be10 edef12663
Author: dreamind <dreamind@gmail.com>
Date:   Wed Oct 21 08:12:16 2020 +1100

    Merge pull request #524 from apromore/feature/AP-2120

    AP-2120 Refine merge model interface

commit edef126638ecb603d336dae6c1a9d0c356853ea5
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Tue Oct 20 23:43:40 2020 +1100

    AP-2120 Update package and classname in apromore-commons

commit 617c9a0d0f0ed757a90bd58207056f2ed90131af
Author: Bruce Nguyen <hoangnguyen7x@yahoo.com>
Date:   Tue Oct 20 19:04:16 2020 +1000

    Fix mispelling: org.joda.time.DateTimeUtilss.

commit 4d6a08d753abec05010844eed3a59f835e49aea5
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Tue Oct 20 00:15:08 2020 +1100

    AP-2120 Refine merge model interface

commit ba4a3be103d73fe84ce6a223c134adc0ce53f90c
Merge: a725fe580 209c4dd72
Author: Simon Raboczi <raboczi@users.noreply.github.com>
Date:   Tue Oct 20 16:41:05 2020 +1000

    Merge pull request #523 from apromore/bug/dbweaving-on-windows/AP-2437

    Bug/dbweaving on windows/ap 2437

commit 209c4dd7201b32aac76c62f61055a5c8a54159e1
Author: Nolan Tellis <nolan.tellis@apromore.com>
Date:   Tue Oct 20 14:43:54 2020 +1100

    Fix static weaving on windows

commit a725fe580662e39a9174dbdb151974c508268fdb
Merge: 513f5b057 f3e3157b2
Author: dreamind <dreamind@gmail.com>
Date:   Tue Oct 20 12:08:35 2020 +1100

    Merge pull request #522 from apromore/AP-2446

    AP-2446 Fix regression in portal search

commit 513f5b057eab20f370f1589aab3af641a6792ce6
Author: Chii Chang <chiichang@users.noreply.github.com>
Date:   Tue Oct 20 11:35:39 2020 +1100

    added unit tests for new filter logics

commit f3e3157b2a1fa46959c973443af5daaa7817c3bc
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Tue Oct 20 10:36:33 2020 +1100

    AP-2446 Fix regression in portal search

commit d127188de85f6f063deef93ef5aa8845cb94966c
Author: Chii Chang <chiichang@users.noreply.github.com>
Date:   Mon Oct 19 17:16:37 2020 +1100

    Reduce the number of points in default chart series

commit 3ab3013c1f7e192513e168fddc21ff4177c3242c
Merge: 2d4fdca75 fe8b9362a
Author: Chii Chang <chiichang@users.noreply.github.com>
Date:   Mon Oct 19 16:11:34 2020 +1100

    Merge pull request #521 from apromore/feature/AP-2432

    AP-2432 Update help and add single variant check for quick filter sho…

commit fe8b9362a7b91aa7bd06f5e7d9cb39027a9a2ac5
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Fri Oct 16 16:55:08 2020 +1100

    AP-2432 Update help and add single variant check for quick filter shortcut in PD

commit 2d4fdca751146724ac325c8dff6abf176ed2127c
Merge: 52a1df6f7 44ee20fbe
Author: frankma <49012776+frank-ma-leonardo@users.noreply.github.com>
Date:   Mon Oct 19 15:17:26 2020 +1100

    Merge pull request #513 from apromore/AP-2304

    Ap 2304 Backend service for sharing artifact (Log, Process, Folder)

commit 52a1df6f75dc8a4203c4ca086c9ab9e26a43f390
Author: Chii Chang <chiichang@users.noreply.github.com>
Date:   Mon Oct 19 15:09:38 2020 +1100

    added new mthod for getting attribute arc duration

commit 44ee20fbe78348ef9d2badf3ee18582b55c61e54
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Mon Oct 19 14:29:08 2020 +1100

    Change method signature

commit ddba89ca0ee40bda4fefff6dd56a5f3f72af9039
Merge: 4906071ff c8f7c5d21
Author: Chii Chang <chiichang@users.noreply.github.com>
Date:   Mon Oct 19 14:28:51 2020 +1100

    Merge pull request #519 from apromore/feature/AP-2432

    AP-2432 Add additional quick filter shortcut in PD

commit c8f7c5d21b76c5d17390626fb53758651c43d419
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Fri Oct 16 16:55:08 2020 +1100

    AP-2432 Add additional quick filter shortcut in PD

commit e02129fc315f679979865739b18bcdab5d4730ed
Author: raboczi <raboczi@icloud.com>
Date:   Mon Oct 19 10:19:54 2020 +1000

    Added a comment noting where to obtain the precompiled Hadoop support files for Windows.  (AP-2416)

commit 52e16e3a1194f2ee4783ea72f3caed2f3538ceeb
Merge: 296499e14 f883deaf0
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Sat Oct 17 11:53:08 2020 +1100

    Rebase

commit 296499e14ef60ce6781666c904abbe63ee4af65b
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Sat Oct 17 11:34:20 2020 +1100

    Fix regressions

commit 83f702e36406af88563017ac9e3d4e888b14ce0f
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Fri Oct 16 18:54:09 2020 +1100

    Reverse IN clause code due to possible incompatibility of spring JPA

commit 4513fde035bda0cf00d186365fa1a7f40d21cc38
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Fri Oct 16 15:22:04 2020 +1100

    Refactor access control APIs out of WorkspaceService to AuthorizationService

commit 8d5afe40690734e2e0d4cc2fe0a770cd9b5273d0
Author: Nolan Tellis <nolan.tellis@apromore.com>
Date:   Thu Oct 15 15:29:02 2020 +1100

    Minor refactoring and test correction

commit b85c9810713fc62956aca82dea0d9af07d409777
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 14 23:46:46 2020 +1100

    comment

commit 9dc003c49ca32f5161c3dd000d3c2c257c1acb85
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 14 09:36:03 2020 +1100

    Fix equal and hashcode methods in Usermetadata POJO

commit 3e377438d1a4ddbf6046a7049c83da36f307f11c
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Tue Oct 13 16:00:32 2020 +1100

    Add unit tests

commit 5779539711d540bcb5d5e43a519ac1494ca39d4f
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Mon Oct 12 11:01:54 2020 +1100

    Feature: CRUD functions for ACL of Log/Process/Folder

commit 347057041923f5166e9ac643468febf5249f01c3
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Thu Oct 8 15:01:16 2020 +1100

    Remove unnecessary cascading in GroupUserMetadata

commit fdb0296a3d2c13c88acf40ba52fe90d9ae39558d
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 7 22:04:18 2020 +1100

    Fix metadata permission after deleting of Log

commit ab793403393907e0c1a0e6e0ae4272dc9934da6e
Author: raboczi <raboczi@icloud.com>
Date:   Fri Oct 16 14:22:13 2020 +1000

    Created TestUtilities.resourceToString to centralize test document processing.

    This should allow documents to be normalized to unix-style line endings for byte-wise comparison in the tests.

commit 4906071ff1f258afc9b7f3f7f485d857e06c2785
Merge: e450496c8 69e0dbe94
Author: dreamind <dreamind@gmail.com>
Date:   Fri Oct 16 15:07:41 2020 +1100

    Merge pull request #518 from apromore/feature/AP-2311-2

    AP-2311 Increase notification hide delay

commit 3aaf13fc4abaad4b3f684c590aa74ec99de0c962
Merge: 638a69401 e450496c8
Author: raboczi <raboczi@icloud.com>
Date:   Fri Oct 16 13:40:05 2020 +1000

    Merge branch 'development' into feature/AP-2416-hadoop-on-windows

commit 638a69401a4a757d302bc8534c43cc11c378c259
Author: Simon Raboczi <simon.raboczi@unimelb.edu.au>
Date:   Thu Oct 15 19:51:04 2020 -0700

    Added hadoop.home.dir system property to CSV Importer logic test environment.

commit 69e0dbe94b70b2d850c82aadd6d15d93271fe220
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Fri Oct 16 13:00:16 2020 +1100

    AP-2311 Increase notification hide delay

commit e450496c80e0e9b03f27891ec013359db9074aad
Merge: 19602996f 456ec72e1
Author: frankma <49012776+frank-ma-leonardo@users.noreply.github.com>
Date:   Fri Oct 16 11:53:24 2020 +1100

    Merge pull request #517 from apromore/AP-2419

    Ap 2419 Add preConditions in LiquidBase changeSet to cover all scenarios

commit 456ec72e1116d0071d1440d1b138abb9ff5215da
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Fri Oct 16 10:52:54 2020 +1100

    Add tableName in indexExists preCondition and change indexName to follow naming conventions

commit cd728d123b40b85d1e8b6f8609fd1c55542742e4
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Thu Oct 15 19:42:44 2020 +1100

    Add preConditions in liquidbase changeSet to cover all scenarios

commit 19602996ff6a6d1a96cde8f64e04b112ef95f6cd
Merge: 0ce682091 546e05b47
Author: Simon Raboczi <raboczi@users.noreply.github.com>
Date:   Thu Oct 15 16:15:17 2020 +1000

    Merge pull request #507 from apromore/feature/AP-2409-enable-checkstyle

    Re-enabled pre-existing Checkstyle code. (AP-2409)

commit 546e05b4731ea77fcc0703b79804c06ee09549fc
Author: raboczi <raboczi@icloud.com>
Date:   Thu Oct 15 16:04:37 2020 +1000

    Incorporated feedback to checkstyle.xml

    - trailing whitespace is permitted if the line is entirely whitespace; this allows Eclipse's auto-indentation
    - documentation links normalized to checkstyle.org

commit f883deaf0e8be5789f7075b78fea9ab5e0971440
Author: Nolan Tellis <nolan.tellis@apromore.com>
Date:   Thu Oct 15 15:29:02 2020 +1100

    Minor refactoring and test correction

commit 98ca9722e0515648a11643855fcb5dac4df8fb3a
Merge: cba9542f9 0ce682091
Author: Nolan Tellis <nolan.tellis@apromore.com>
Date:   Thu Oct 15 14:44:28 2020 +1100

    Merge remote-tracking branch 'origin/development' into AP-2304

commit 0ce682091adb3881998758ed9f0572a1c2bb3179
Merge: d5db52edb 1c93e1f9b
Author: Simon Raboczi <raboczi@users.noreply.github.com>
Date:   Thu Oct 15 12:00:21 2020 +1000

    Merge pull request #516 from apromore/bugfix/apromore-commons

    Fixed build failure in Apromore-Portal

commit 1c93e1f9bb7681fe6a881c147fe113616576a147
Author: raboczi <raboczi@icloud.com>
Date:   Thu Oct 15 11:53:42 2020 +1000

    Fixed build failure in Apromore-Portal pom.xml: commons dependency should be apromore-commons

commit d5db52edb88fdc9a3af7d3c8206b7cd41a9f2f5f
Merge: d70f308e2 5715b700a
Author: Simon Raboczi <raboczi@users.noreply.github.com>
Date:   Thu Oct 15 11:04:03 2020 +1000

    Merge pull request #515 from apromore/bugfix/create-h2

    Modified "ant create-h2" for Liquibase compatibility.

commit 38aa01e1ea2cfbc09de7ab366c51776ae6a88ba2
Merge: a8be97cf1 5715b700a
Author: raboczi <raboczi@icloud.com>
Date:   Thu Oct 15 11:03:03 2020 +1000

    Merge branch 'bugfix/create-h2' into feature/AP-2416-hadoop-on-windows

commit 5715b700a51b6030e782683712e45601f1e5f3eb
Author: raboczi <raboczi@icloud.com>
Date:   Thu Oct 15 10:38:42 2020 +1000

    Modified "ant create-h2" for Liquibase compatibility.

commit d70f308e24adfe2131b058d9105d730fb2428177
Merge: 4f73c0aa1 3eb998e4f
Author: dreamind <dreamind@gmail.com>
Date:   Thu Oct 15 10:17:16 2020 +1100

    Merge pull request #514 from apromore/feature/AP-1160-2

    AP-1160 Commons source code formatting

commit 3eb998e4f5ada60ca29c41a54c3b378054b269cc
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Thu Oct 15 10:16:03 2020 +1100

    AP-1160 Commons source code formatting

commit 4f73c0aa1c1b84bc4781c9063b205b769d76bb6e
Merge: 3c4afd11e ccb80c695
Author: dreamind <dreamind@gmail.com>
Date:   Thu Oct 15 09:13:35 2020 +1100

    Merge pull request #509 from apromore/feature/AP-1160

    AP-1160 Refactor date, time and duration handling

commit cba9542f9a8fce48708af76cc3db445bc9dd38a9
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 14 23:46:46 2020 +1100

    comment

commit ec444c96534ac36370b8a2aff48780660a7d10b6
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 14 09:36:03 2020 +1100

    Fix equal and hashcode methods in Usermetadata POJO

commit eaa3f00180aa64c68120073328dfd3f5f5b7e868
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Tue Oct 13 16:00:32 2020 +1100

    Add unit tests

commit be2cdb4ffe24be5801ff252da337e15340f15491
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Mon Oct 12 11:01:54 2020 +1100

    Feature: CRUD functions for ACL of Log/Process/Folder

commit 22e2d4aac1e7954dde361caf96257d017256987f
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Thu Oct 8 15:01:16 2020 +1100

    Remove unnecessary cascading in GroupUserMetadata

commit edaddb63f2275690c2ba9c6ab1f4d432dbd29b15
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 7 22:04:18 2020 +1100

    Fix metadata permission after deleting of Log

commit a8be97cf181d409c8213c2c9956b3541456d605e
Author: raboczi <raboczi@icloud.com>
Date:   Wed Oct 14 16:37:46 2020 +1000

    Fixed "ant create-h2" target, broken when Apromore-Database relocated the SQL scripts.

commit 3c4afd11e3876ec3716d48f8c246b1d4016a906e
Merge: 0c5bbf447 644c7ebfa
Author: frankma <49012776+frank-ma-leonardo@users.noreply.github.com>
Date:   Wed Oct 14 16:11:51 2020 +1100

    Merge pull request #512 from apromore/AP-2419

    AP-2419 Fix regressions in user metadata features

commit ccb80c695031998d293634cab59df742137486dd
Author: Nolan Tellis <nolan.tellis@apromore.com>
Date:   Wed Oct 14 15:27:10 2020 +1100

    Minor refactoring to remove unwanted loops

commit c0478dc29fe3f90f4a22cb6209efe73292727404
Author: raboczi <raboczi@icloud.com>
Date:   Wed Oct 14 12:57:30 2020 +1000

    Added the latest precompiled native Windows libraries to support Hadoop (AP-2416)

    Beware that the latest precompiled native libraries are 3.2.1, whereas Apromore is using Hadoop 3.3.0.

commit 644c7ebfa81c3211484c2f6da2f64237e86a1d11
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 14 13:32:05 2020 +1100

    AP-2419 Fix regressions in user metadata features

commit 0c5bbf447cec98860d8ac9a76c2a0a893357b25a
Merge: b6b80d5c0 06cac7f38
Author: dreamind <dreamind@gmail.com>
Date:   Wed Oct 14 12:39:24 2020 +1100

    Merge pull request #511 from apromore/feature/AP-2244-2

    AP-2244 Check only one selection for rename

commit abfe339d9774f26102ac57a8450d4602715af0fc
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Mon Oct 12 14:54:31 2020 +1100

    AP-1160 Refactor date, time and duration handling

commit 06cac7f383a5f4caba231ab850524248d212934b
Author: Nolan Tellis <nolan.tellis@apromore.com>
Date:   Wed Oct 14 11:18:47 2020 +1100

    Minor changes in code function

commit a526e56219342f1d4aeb0e94472c5788a32e6c10
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Tue Oct 13 17:23:12 2020 +1100

    AP-2244 Check only one selection for rename

commit f7a19e118bf1ba193dad3d9202c7be4e19204c7f
Author: raboczi <raboczi@icloud.com>
Date:   Mon Oct 12 16:34:22 2020 +1000

    Re-enabled pre-existing Checkstyle code. (AP-2409)

    Updated Checkstyle from v2 to v3; the old (partially incompatible checkstyle XML is retained as checkstyle-v2.xml).
    The v3 (checkstyle.xml) is mostly default; Javadoc, DesignForExtension, and OperatorWrap are notable changes.

Commit:
Update bpmn modeler

Commit:
Update bpmn modeler

Commit:
Update bpmn modeler

Commit:
Update bpmn modeler

Commit:
Fix unidentified validation errors

Commit:
Fix unidentified validation errors

Commit:
Fix unidentified validation errors

Commit:
Fix unidentified validation errors

Commit:
Fix NaN distribution value

Commit:
Fix NaN distribution value

Commit:
Fix NaN distribution value

Commit:
Fix NaN distribution value

Commit:
Squashed commit of the following:

commit 21bfd788e89d496d7bf8801d739a96d50c09d1bb
Merge: 4d313eead 90ab32199
Author: dreamind <dreamind@gmail.com>
Date:   Wed Oct 28 00:16:04 2020 +1100

    Merge pull request #528 from apromore/feature/AP-2303

    AP-2303 Share Dialog

commit 90ab32199bccb27fb21908713eace6ea4df611ec
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Tue Oct 27 23:52:27 2020 +1100

    AP-2303 Disable folder sharing and check remaining/duplicates

commit 125245ec902867e9a027f519223687d2237d6d28
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Mon Oct 26 18:47:20 2020 +1100

    Fix copyright headers

commit 1cc805718e8f8186a723a1f42499a62750b82854
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Wed Sep 23 17:45:02 2020 +1000

    AP-2303 Share Dialog

commit 4d313eeadae718aa941da2acbe26d6a04c5885ff
Merge: ca1747d11 caeda214c
Author: Simon Raboczi <raboczi@users.noreply.github.com>
Date:   Tue Oct 27 13:35:29 2020 +1000

    Merge pull request #530 from apromore/feature/AP-2300

    fixed encoding

commit ca1747d11319fc50e6c5729f3d274b12d88e667a
Merge: 92c91f8cf 688f70299
Author: Simon Raboczi <raboczi@users.noreply.github.com>
Date:   Tue Oct 27 12:57:06 2020 +1000

    Merge pull request #531 from apromore/feature/AP-2384-refactor-OSGi-manifest

    Feature/AP-2384 Refactor OSGi manifest

commit 688f702993e4d8d72b7aa3704353bbc7e7d72d11
Author: raboczi <raboczi@icloud.com>
Date:   Mon Oct 26 22:23:46 2020 +1000

    Removed remaining OSGi Import-Bundle manifest headers. (AP-2384)

commit fad0cfa0c809d2088449df62af080cfece956af3
Author: raboczi <raboczi@icloud.com>
Date:   Mon Oct 26 16:53:20 2020 +1000

    Removed use of OSGi Import-Bundle from Apromore-Manager (AP-2384)

commit 92c91f8cf8b387c33778439f969671325dbd5701
Merge: db95b0056 a33eebbe3
Author: dreamind <dreamind@gmail.com>
Date:   Sat Oct 24 20:57:01 2020 +1100

    Merge pull request #529 from apromore/feature/AP-2120-2432

    AP-2120-2432 Merge and PD shortcuts fixes

commit a33eebbe39fe639db160d450d9e4fa68b06c8d06
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Sat Oct 24 20:55:30 2020 +1100

    AP-2120-2432 Merge and PD shortcuts fixes

commit db95b00566b9675c2c89c82bc0b1cc9b9506fcc8
Merge: f39f5ebea 6812f38a4
Author: frankma <49012776+frank-ma-leonardo@users.noreply.github.com>
Date:   Thu Oct 22 12:29:04 2020 +1100

    Merge pull request #525 from apromore/AP-2431

    APIs to get dependent artifact of user metadata

commit 6812f38a48031ccc5b839dc43cd026cf04dc85d2
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Thu Oct 22 09:49:16 2020 +1100

    Add method to get the usermetadata AccessType

commit 1e4e08b88ffe540b4d2c98c4fd2b2dab130ec16d
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 21 17:41:39 2020 +1100

    Remove unnecessary assertions

commit f39f5ebea69de65982ee0a06274c40ffe73cd06e
Merge: 41e0d2198 e02129fc3
Author: raboczi <raboczi@icloud.com>
Date:   Wed Oct 21 15:12:05 2020 +1000

    Merge branch 'feature/AP-2416-hadoop-on-windows' into development

commit 41e0d2198619db71302aa74d7a5fc17c3d1e4cf6
Merge: 915b7bf09 50f7b6c95
Author: dreamind <dreamind@gmail.com>
Date:   Wed Oct 21 11:22:48 2020 +1100

    Merge pull request #526 from apromore/feature/AP-2236

    AP-2236 Fix long graph search result in PD

commit 50f7b6c957df9a4bdf55620540ad35009ec9dbec
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Wed Oct 21 11:21:22 2020 +1100

    AP-2236 Fix long graph search result in PD

commit a1ecf9ff415bf86a065214c145ae089f7d728afc
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 21 09:06:13 2020 +1100

    APIs to get dependent artifact of user metadata

commit 915b7bf095e6c9f547d162ff5d077bc4af6e2959
Merge: ba4a3be10 edef12663
Author: dreamind <dreamind@gmail.com>
Date:   Wed Oct 21 08:12:16 2020 +1100

    Merge pull request #524 from apromore/feature/AP-2120

    AP-2120 Refine merge model interface

commit edef126638ecb603d336dae6c1a9d0c356853ea5
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Tue Oct 20 23:43:40 2020 +1100

    AP-2120 Update package and classname in apromore-commons

commit 617c9a0d0f0ed757a90bd58207056f2ed90131af
Author: Bruce Nguyen <hoangnguyen7x@yahoo.com>
Date:   Tue Oct 20 19:04:16 2020 +1000

    Fix mispelling: org.joda.time.DateTimeUtilss.

commit 4d6a08d753abec05010844eed3a59f835e49aea5
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Tue Oct 20 00:15:08 2020 +1100

    AP-2120 Refine merge model interface

commit ba4a3be103d73fe84ce6a223c134adc0ce53f90c
Merge: a725fe580 209c4dd72
Author: Simon Raboczi <raboczi@users.noreply.github.com>
Date:   Tue Oct 20 16:41:05 2020 +1000

    Merge pull request #523 from apromore/bug/dbweaving-on-windows/AP-2437

    Bug/dbweaving on windows/ap 2437

commit 209c4dd7201b32aac76c62f61055a5c8a54159e1
Author: Nolan Tellis <nolan.tellis@apromore.com>
Date:   Tue Oct 20 14:43:54 2020 +1100

    Fix static weaving on windows

commit a725fe580662e39a9174dbdb151974c508268fdb
Merge: 513f5b057 f3e3157b2
Author: dreamind <dreamind@gmail.com>
Date:   Tue Oct 20 12:08:35 2020 +1100

    Merge pull request #522 from apromore/AP-2446

    AP-2446 Fix regression in portal search

commit 513f5b057eab20f370f1589aab3af641a6792ce6
Author: Chii Chang <chiichang@users.noreply.github.com>
Date:   Tue Oct 20 11:35:39 2020 +1100

    added unit tests for new filter logics

commit f3e3157b2a1fa46959c973443af5daaa7817c3bc
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Tue Oct 20 10:36:33 2020 +1100

    AP-2446 Fix regression in portal search

commit d127188de85f6f063deef93ef5aa8845cb94966c
Author: Chii Chang <chiichang@users.noreply.github.com>
Date:   Mon Oct 19 17:16:37 2020 +1100

    Reduce the number of points in default chart series

commit 3ab3013c1f7e192513e168fddc21ff4177c3242c
Merge: 2d4fdca75 fe8b9362a
Author: Chii Chang <chiichang@users.noreply.github.com>
Date:   Mon Oct 19 16:11:34 2020 +1100

    Merge pull request #521 from apromore/feature/AP-2432

    AP-2432 Update help and add single variant check for quick filter sho…

commit fe8b9362a7b91aa7bd06f5e7d9cb39027a9a2ac5
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Fri Oct 16 16:55:08 2020 +1100

    AP-2432 Update help and add single variant check for quick filter shortcut in PD

commit 2d4fdca751146724ac325c8dff6abf176ed2127c
Merge: 52a1df6f7 44ee20fbe
Author: frankma <49012776+frank-ma-leonardo@users.noreply.github.com>
Date:   Mon Oct 19 15:17:26 2020 +1100

    Merge pull request #513 from apromore/AP-2304

    Ap 2304 Backend service for sharing artifact (Log, Process, Folder)

commit 52a1df6f75dc8a4203c4ca086c9ab9e26a43f390
Author: Chii Chang <chiichang@users.noreply.github.com>
Date:   Mon Oct 19 15:09:38 2020 +1100

    added new mthod for getting attribute arc duration

commit 44ee20fbe78348ef9d2badf3ee18582b55c61e54
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Mon Oct 19 14:29:08 2020 +1100

    Change method signature

commit ddba89ca0ee40bda4fefff6dd56a5f3f72af9039
Merge: 4906071ff c8f7c5d21
Author: Chii Chang <chiichang@users.noreply.github.com>
Date:   Mon Oct 19 14:28:51 2020 +1100

    Merge pull request #519 from apromore/feature/AP-2432

    AP-2432 Add additional quick filter shortcut in PD

commit c8f7c5d21b76c5d17390626fb53758651c43d419
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Fri Oct 16 16:55:08 2020 +1100

    AP-2432 Add additional quick filter shortcut in PD

commit e02129fc315f679979865739b18bcdab5d4730ed
Author: raboczi <raboczi@icloud.com>
Date:   Mon Oct 19 10:19:54 2020 +1000

    Added a comment noting where to obtain the precompiled Hadoop support files for Windows.  (AP-2416)

commit 52e16e3a1194f2ee4783ea72f3caed2f3538ceeb
Merge: 296499e14 f883deaf0
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Sat Oct 17 11:53:08 2020 +1100

    Rebase

commit 296499e14ef60ce6781666c904abbe63ee4af65b
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Sat Oct 17 11:34:20 2020 +1100

    Fix regressions

commit 83f702e36406af88563017ac9e3d4e888b14ce0f
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Fri Oct 16 18:54:09 2020 +1100

    Reverse IN clause code due to possible incompatibility of spring JPA

commit 4513fde035bda0cf00d186365fa1a7f40d21cc38
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Fri Oct 16 15:22:04 2020 +1100

    Refactor access control APIs out of WorkspaceService to AuthorizationService

commit 8d5afe40690734e2e0d4cc2fe0a770cd9b5273d0
Author: Nolan Tellis <nolan.tellis@apromore.com>
Date:   Thu Oct 15 15:29:02 2020 +1100

    Minor refactoring and test correction

commit b85c9810713fc62956aca82dea0d9af07d409777
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 14 23:46:46 2020 +1100

    comment

commit 9dc003c49ca32f5161c3dd000d3c2c257c1acb85
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 14 09:36:03 2020 +1100

    Fix equal and hashcode methods in Usermetadata POJO

commit 3e377438d1a4ddbf6046a7049c83da36f307f11c
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Tue Oct 13 16:00:32 2020 +1100

    Add unit tests

commit 5779539711d540bcb5d5e43a519ac1494ca39d4f
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Mon Oct 12 11:01:54 2020 +1100

    Feature: CRUD functions for ACL of Log/Process/Folder

commit 347057041923f5166e9ac643468febf5249f01c3
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Thu Oct 8 15:01:16 2020 +1100

    Remove unnecessary cascading in GroupUserMetadata

commit fdb0296a3d2c13c88acf40ba52fe90d9ae39558d
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 7 22:04:18 2020 +1100

    Fix metadata permission after deleting of Log

commit ab793403393907e0c1a0e6e0ae4272dc9934da6e
Author: raboczi <raboczi@icloud.com>
Date:   Fri Oct 16 14:22:13 2020 +1000

    Created TestUtilities.resourceToString to centralize test document processing.

    This should allow documents to be normalized to unix-style line endings for byte-wise comparison in the tests.

commit 4906071ff1f258afc9b7f3f7f485d857e06c2785
Merge: e450496c8 69e0dbe94
Author: dreamind <dreamind@gmail.com>
Date:   Fri Oct 16 15:07:41 2020 +1100

    Merge pull request #518 from apromore/feature/AP-2311-2

    AP-2311 Increase notification hide delay

commit 3aaf13fc4abaad4b3f684c590aa74ec99de0c962
Merge: 638a69401 e450496c8
Author: raboczi <raboczi@icloud.com>
Date:   Fri Oct 16 13:40:05 2020 +1000

    Merge branch 'development' into feature/AP-2416-hadoop-on-windows

commit 638a69401a4a757d302bc8534c43cc11c378c259
Author: Simon Raboczi <simon.raboczi@unimelb.edu.au>
Date:   Thu Oct 15 19:51:04 2020 -0700

    Added hadoop.home.dir system property to CSV Importer logic test environment.

commit 69e0dbe94b70b2d850c82aadd6d15d93271fe220
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Fri Oct 16 13:00:16 2020 +1100

    AP-2311 Increase notification hide delay

commit e450496c80e0e9b03f27891ec013359db9074aad
Merge: 19602996f 456ec72e1
Author: frankma <49012776+frank-ma-leonardo@users.noreply.github.com>
Date:   Fri Oct 16 11:53:24 2020 +1100

    Merge pull request #517 from apromore/AP-2419

    Ap 2419 Add preConditions in LiquidBase changeSet to cover all scenarios

commit 456ec72e1116d0071d1440d1b138abb9ff5215da
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Fri Oct 16 10:52:54 2020 +1100

    Add tableName in indexExists preCondition and change indexName to follow naming conventions

commit cd728d123b40b85d1e8b6f8609fd1c55542742e4
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Thu Oct 15 19:42:44 2020 +1100

    Add preConditions in liquidbase changeSet to cover all scenarios

commit 19602996ff6a6d1a96cde8f64e04b112ef95f6cd
Merge: 0ce682091 546e05b47
Author: Simon Raboczi <raboczi@users.noreply.github.com>
Date:   Thu Oct 15 16:15:17 2020 +1000

    Merge pull request #507 from apromore/feature/AP-2409-enable-checkstyle

    Re-enabled pre-existing Checkstyle code. (AP-2409)

commit 546e05b4731ea77fcc0703b79804c06ee09549fc
Author: raboczi <raboczi@icloud.com>
Date:   Thu Oct 15 16:04:37 2020 +1000

    Incorporated feedback to checkstyle.xml

    - trailing whitespace is permitted if the line is entirely whitespace; this allows Eclipse's auto-indentation
    - documentation links normalized to checkstyle.org

commit f883deaf0e8be5789f7075b78fea9ab5e0971440
Author: Nolan Tellis <nolan.tellis@apromore.com>
Date:   Thu Oct 15 15:29:02 2020 +1100

    Minor refactoring and test correction

commit 98ca9722e0515648a11643855fcb5dac4df8fb3a
Merge: cba9542f9 0ce682091
Author: Nolan Tellis <nolan.tellis@apromore.com>
Date:   Thu Oct 15 14:44:28 2020 +1100

    Merge remote-tracking branch 'origin/development' into AP-2304

commit 0ce682091adb3881998758ed9f0572a1c2bb3179
Merge: d5db52edb 1c93e1f9b
Author: Simon Raboczi <raboczi@users.noreply.github.com>
Date:   Thu Oct 15 12:00:21 2020 +1000

    Merge pull request #516 from apromore/bugfix/apromore-commons

    Fixed build failure in Apromore-Portal

commit 1c93e1f9bb7681fe6a881c147fe113616576a147
Author: raboczi <raboczi@icloud.com>
Date:   Thu Oct 15 11:53:42 2020 +1000

    Fixed build failure in Apromore-Portal pom.xml: commons dependency should be apromore-commons

commit d5db52edb88fdc9a3af7d3c8206b7cd41a9f2f5f
Merge: d70f308e2 5715b700a
Author: Simon Raboczi <raboczi@users.noreply.github.com>
Date:   Thu Oct 15 11:04:03 2020 +1000

    Merge pull request #515 from apromore/bugfix/create-h2

    Modified "ant create-h2" for Liquibase compatibility.

commit 38aa01e1ea2cfbc09de7ab366c51776ae6a88ba2
Merge: a8be97cf1 5715b700a
Author: raboczi <raboczi@icloud.com>
Date:   Thu Oct 15 11:03:03 2020 +1000

    Merge branch 'bugfix/create-h2' into feature/AP-2416-hadoop-on-windows

commit 5715b700a51b6030e782683712e45601f1e5f3eb
Author: raboczi <raboczi@icloud.com>
Date:   Thu Oct 15 10:38:42 2020 +1000

    Modified "ant create-h2" for Liquibase compatibility.

commit d70f308e24adfe2131b058d9105d730fb2428177
Merge: 4f73c0aa1 3eb998e4f
Author: dreamind <dreamind@gmail.com>
Date:   Thu Oct 15 10:17:16 2020 +1100

    Merge pull request #514 from apromore/feature/AP-1160-2

    AP-1160 Commons source code formatting

commit 3eb998e4f5ada60ca29c41a54c3b378054b269cc
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Thu Oct 15 10:16:03 2020 +1100

    AP-1160 Commons source code formatting

commit 4f73c0aa1c1b84bc4781c9063b205b769d76bb6e
Merge: 3c4afd11e ccb80c695
Author: dreamind <dreamind@gmail.com>
Date:   Thu Oct 15 09:13:35 2020 +1100

    Merge pull request #509 from apromore/feature/AP-1160

    AP-1160 Refactor date, time and duration handling

commit cba9542f9a8fce48708af76cc3db445bc9dd38a9
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 14 23:46:46 2020 +1100

    comment

commit ec444c96534ac36370b8a2aff48780660a7d10b6
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 14 09:36:03 2020 +1100

    Fix equal and hashcode methods in Usermetadata POJO

commit eaa3f00180aa64c68120073328dfd3f5f5b7e868
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Tue Oct 13 16:00:32 2020 +1100

    Add unit tests

commit be2cdb4ffe24be5801ff252da337e15340f15491
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Mon Oct 12 11:01:54 2020 +1100

    Feature: CRUD functions for ACL of Log/Process/Folder

commit 22e2d4aac1e7954dde361caf96257d017256987f
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Thu Oct 8 15:01:16 2020 +1100

    Remove unnecessary cascading in GroupUserMetadata

commit edaddb63f2275690c2ba9c6ab1f4d432dbd29b15
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 7 22:04:18 2020 +1100

    Fix metadata permission after deleting of Log

commit a8be97cf181d409c8213c2c9956b3541456d605e
Author: raboczi <raboczi@icloud.com>
Date:   Wed Oct 14 16:37:46 2020 +1000

    Fixed "ant create-h2" target, broken when Apromore-Database relocated the SQL scripts.

commit 3c4afd11e3876ec3716d48f8c246b1d4016a906e
Merge: 0c5bbf447 644c7ebfa
Author: frankma <49012776+frank-ma-leonardo@users.noreply.github.com>
Date:   Wed Oct 14 16:11:51 2020 +1100

    Merge pull request #512 from apromore/AP-2419

    AP-2419 Fix regressions in user metadata features

commit ccb80c695031998d293634cab59df742137486dd
Author: Nolan Tellis <nolan.tellis@apromore.com>
Date:   Wed Oct 14 15:27:10 2020 +1100

    Minor refactoring to remove unwanted loops

commit c0478dc29fe3f90f4a22cb6209efe73292727404
Author: raboczi <raboczi@icloud.com>
Date:   Wed Oct 14 12:57:30 2020 +1000

    Added the latest precompiled native Windows libraries to support Hadoop (AP-2416)

    Beware that the latest precompiled native libraries are 3.2.1, whereas Apromore is using Hadoop 3.3.0.

commit 644c7ebfa81c3211484c2f6da2f64237e86a1d11
Author: frank-ma-leonardo <f.ma@leonardo.com.au>
Date:   Wed Oct 14 13:32:05 2020 +1100

    AP-2419 Fix regressions in user metadata features

commit 0c5bbf447cec98860d8ac9a76c2a0a893357b25a
Merge: b6b80d5c0 06cac7f38
Author: dreamind <dreamind@gmail.com>
Date:   Wed Oct 14 12:39:24 2020 +1100

    Merge pull request #511 from apromore/feature/AP-2244-2

    AP-2244 Check only one selection for rename

commit abfe339d9774f26102ac57a8450d4602715af0fc
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Mon Oct 12 14:54:31 2020 +1100

    AP-1160 Refactor date, time and duration handling

commit 06cac7f383a5f4caba231ab850524248d212934b
Author: Nolan Tellis <nolan.tellis@apromore.com>
Date:   Wed Oct 14 11:18:47 2020 +1100

    Minor changes in code function

commit a526e56219342f1d4aeb0e94472c5788a32e6c10
Author: Ivo Widjaja <dreamind@gmail.com>
Date:   Tue Oct 13 17:23:12 2020 +1100

    AP-2244 Check only one selection for rename

commit caeda214c134b79d06f3cf86eb96df135aeae0a6
Author: Own Daghagheleh <>
Date:   Tue Oct 13 14:05:30 2020 +1100

    fixed encoding

commit f7a19e118bf1ba193dad3d9202c7be4e19204c7f
Author: raboczi <raboczi@icloud.com>
Date:   Mon Oct 12 16:34:22 2020 +1000

    Re-enabled pre-existing Checkstyle code. (AP-2409)

    Updated Checkstyle from v2 to v3; the old (partially incompatible checkstyle XML is retained as checkstyle-v2.xml).
    The v3 (checkstyle.xml) is mostly default; Javadoc, DesignForExtension, and OperatorWrap are notable changes.

Commit:
* Corrected documentation regarding database.
* Corrected code to include new liquibase user.

Commit:
Correcting test property file

Commit:
Merge remote-tracking branch 'origin/feature/AP-1874' into feature/AP-1874

# Conflicts:
#	Apromore-Custom-Plugins/Merge-Portal-Plugin/src/main/resources/zul/processmerge.zul
#	Apromore-Custom-Plugins/Process-Discoverer-Portal-Plugin/src/main/resources/processdiscoverer/zul/processDiscoverer.zul

Commit:
Merge remote-tracking branch 'origin/feature/AP-1874' into feature/AP-1874

Commit:
corrected root to apromore

Commit:
Minor changes in liquibase user

raboczi commented:
The show-stopping problem when I build this branch is that the BPMN editor's "Save" and "Save As" toolbar buttons stop working.

The simulation parameter panel works nicely, although (like the rest of the system) it's a little awkward to use in dark mode on macOS.  (Not a problem that needs addressing any time soon.)

Commit:
Corrected Readme docs

raboczi committed:
Update database-jpa.xml

Two elements with same @id ("dataSource") in the same XML file is confusing, so removed it from the embedded bean.

Commit:
Merge remote-tracking branch 'origin/task/update-liquibase-doc/AP-2484' into task/update-liquibase-doc/AP-2484

nolantellis committed:
Merge pull request #534 from apromore/task/update-liquibase-doc/AP-2484

* Corrected documentation regarding database.

Commit:
Fix AP-2493

Commit:
Fix AP-2493

chiichang committed:
AP-2474

raboczi committed:
Merge pull request #538 from apromore/feature/apmlog

Feature/apmlog

Commit:
Add useSimulationPanel attribute

Commit:
Merge remote-tracking branch 'origin/feature/AP-1874' into feature/AP-1874

# Conflicts:
#	Apromore-Core-Components/Apromore-BPMNEditor/src/main/scripts/editor.js
#	Apromore-Core-Components/Apromore-BPMNEditor/src/main/scripts/plugins/simulationPanel.js
